src/codelens/TestCodeLensProvider.ts

```diff
@@ -26,14 +30,25 @@ export class TestCodeLensProvider implements CodeLensProvider, Disposable {
         this.onDidChangeCodeLensesEmitter.fire();
     }
 
-    public async provideCodeLenses(document: TextDocument, _token: CancellationToken): Promise<CodeLens[]> {
+    public async provideCodeLenses(document: TextDocument, token: CancellationToken): Promise<CodeLens[]> {
         if (!this.isActivated) {
             return [];
         }
 
         try {
-            const items: ITestItem[] = await testItemModel.getItemsForCodeLens(document.uri);
-            return this.getCodeLenses(items);
+            const timeout: number = this.getRequestDelay(document.uri);
+            return new Promise<CodeLens[]>((resolve: (ids: CodeLens[]) => void): void => {
+                token.onCancellationRequested(() => {
+                    clearTimeout(timeoutHandle);
+                    resolve([]);
+                    this.dispose();
```

Comment:
what's the purpose to dispose here?

Reply:
Yes this seems can be removed. (Checked at the VS Code repo about the usage of this, and not found such kind of usage)